lnwire/partial_sig.go

```diff
@@ -0,0 +1,197 @@
+package lnwire
+
+import (
+	"io"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr/musig2"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+const (
+	// PartialSigLen is the length of a musig2 partial signature.
+	PartialSigLen = 32
+
+	// PartialSigRecordType is the type of the tlv record for a musig2
+	// partial signature. This is an _even_ type, which means it's required
+	// if included.
+	PartialSigRecordType = 6
```

Comment:
Feels like all these types should be declared in 1 spot so it is easy to see which types are taken etc? cause wont the type number maybe vary depending on which message it is in?

Also, should we be strict and force the type here to do `=  tlv.Type(4)` rather? (same for `NonceRecordType`)

Reply:
> Feels like all these types should be declared in 1 spot so it is easy to see which types are taken etc? cause wont the type number maybe vary depending on which message it is in?

Yeah good question, technically the TLV namespace is sort of scoped to each individual message. Eg: all the mesages can use a field of 1 and it won't collide since the message itself serves as a local namespace. In the current spec draft, we end up creating a psuedo-global namespace that applies to all these existing messages. 

I modified the declaration to declare the type `tlv.Type = 4`.